### PR TITLE
prime repo

### PR DESCRIPTION
**Pull request by deads2k**
adds makefile, glide.yaml, and bump to get started.

### CODE REVIEW

**Comment by squeed** [comment]
cc @danwinship

**Comment by deads2k** [comment]
We plan to scale back CI to unit tests, verifies, and building since no images from here are used yet. Then we'll add images and CI for them back as we go.

**Comment by sttts** [comment]
/hold

until we have disable publishing, and CI + image is near to be in place

File: images/network-controller/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,12 @@
+FROM registry.svc.ci.openshift.org/openshift/builder:golang-1.12 AS builder
+WORKDIR /go/src/github.com/openshift/sdn
+COPY . .
+RUN make build --warn-undefined-variables WHAT=cmd/network-controller
```
Comment (tnozicka) [inline images/network-controller/Dockerfile]: WHAT isn't supported
I think there is a separate target for every binary hopefully, try `make help`

File: images/network-controller/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+FROM centos:7
+COPY --from=builder /go/src/github.com/openshift/sdn/network-controller /usr/bin/
+ENTRYPOINT ["/usr/bin/network-controller"]
+LABEL io.k8s.display-name="OpenShift Network Controller" \
+      io.k8s.description="This is a component of OpenShift, it controls the network" \
+      com.redhat.component="network-controller" \
+      io.openshift.tags="openshift,network-controller,sdn"
```
Comment (tnozicka) [inline images/network-controller/Dockerfile]: line end

**Comment by tnozicka** [comment]
I'll fix the CI to take correct paths for those Dockerfiles

File: images/network-controller/Dockerfile (hunk 1 above)
Comment (sttts) [reply images/network-controller/Dockerfile]: k

File: images/network-controller/Dockerfile (hunk 2 above)
Comment (sttts) [reply images/network-controller/Dockerfile]: k

File: Makefile [hunk 3]
```diff
@@ -14,7 +14,8 @@ include $(addprefix ./vendor/github.com/openshift/library-go/alpha-build-machine
 # $2 - Dockerfile path
 # $3 - context directory for image build
 # It will generate target "image-$(1)" for builing the image an binding it as a prerequisite to target "images".
-$(call build-image,origin-$(GO_PACKAGE),./images/sdn/Dockerfile,.)
+$(call build-image,origin-sdn,./images/sdn/Dockerfile,.)
```
Comment (tnozicka) [inline Makefile]: reflect the real names here pls so we avoid confusion

Comment (sttts) [reply Makefile]: done

**Comment by sttts** [comment]
Waiting for network-controller to be published.

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+)
+
+# This will call a macro called "build-image" which will generate image specific targets based on the parameters:
+# $0 - macro name
+# $1 - target suffix
+# $2 - Dockerfile path
+# $3 - context directory for image build
+# It will generate target "image-$(1)" for builing the image an binding it as a prerequisite to target "images".
+$(call build-image,origin-node,./images/sdn/Dockerfile,.)
+$(call build-image,origin-network-controller,./images/network-controller/Dockerfile,.)
```
Comment (danwinship) [inline Makefile]: Neither of these image names is really right; the sdn node image is currently called "origin-node", but that's a historical accident. (The SDN binary is the only thing left in the old `openshift start node` image.) It should be renamed to "origin-sdn" or "origin-openshift-sdn" while we're moving it.

For the controller side, "origin-network-controller" makes it sound generically network-related, but it's not; it's very specifically the controller for just the openshift-sdn network plugin. (It doesn't get run if you have any other network plugin configured.) So it should be "origin-sdn-controller" or "origin-openshift-sdn-controller".

Comment (danwinship) [reply Makefile]: (comments about "network controller" vs "sdn controller" apply to the labels inside the image too, though those are easier to fix later without needing to coordinate multiple repos).

Also, you could just merge them into a single image if that made sense...

Comment (deads2k) [reply Makefile]: -node is legacy, we aren't planning to change that as part of this push.


I suspect as a post-step, you'll want a single unified sdn image, but I think that can happen just as easily afterwards.  This is about seeing if we can get fast enough responses on images to make your CI a little more effective.

Comment (sttts) [reply Makefile]: I agree with the rename of the node image. But let's leave that for a follow-up. We cannot safe the world in one PR. Better step by step.

@stlaz see network-controller name.

@danwinship is that a definitive answer for the network-controller image name?

File: images/node/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,39 @@
+FROM registry.svc.ci.openshift.org/openshift/builder:golang-1.12 AS builder
+WORKDIR /go/src/github.com/openshift/sdn
+COPY . .
+RUN go build "vendor/github.com/containernetworking/plugins/plugins/ipam/host-local"
+RUN go build "vendor/github.com/containernetworking/plugins/plugins/main/loopback"
+RUN mkdir -p /tmp/build; \
+    cp /go/src/github.com/openshift/origin/_output/local/bin/linux/$(go env GOARCH)/openshift-sdn /tmp/build/openshift-sdn; \
```
Comment (tnozicka) [inline images/node/Dockerfile]: this won't match

File: images/node/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,39 @@
+FROM registry.svc.ci.openshift.org/openshift/builder:golang-1.12 AS builder
+WORKDIR /go/src/github.com/openshift/sdn
+COPY . .
+RUN go build "vendor/github.com/containernetworking/plugins/plugins/ipam/host-local"
```
Comment (tnozicka) [inline images/node/Dockerfile]: do `make build` here

File: Makefile (hunk 4 above)
Comment (tnozicka) [reply Makefile]: we need to change it in CI as that's where  the definitive name for CI resides. "origin-sdn-controller" or "origin-openshift-sdn-controller"?

Comment (sttts) [reply Makefile]: from Slack discussion with @danwinship: sdn-controller (with automatic ose/origin prefix)

File: images/node/Dockerfile (hunk 6 above)
Comment (sttts) [reply images/node/Dockerfile]: fixed.

File: images/node/Dockerfile (hunk 5 above)
Comment (tnozicka) [reply images/node/Dockerfile]: @sttts this break the build

Comment (tnozicka) [reply images/node/Dockerfile]:
```
COPY --from=builder /go/src/github.com/openshift/sdn/openshift-sdn /usr/bin/
```

Comment (tnozicka) [reply images/node/Dockerfile]: or pick this file https://github.com/openshift/origin/pull/23273/files#diff-29391d0364ab2ca3cb45701b081dcf3aR1

**Pull request merged**